packages/backend/src/QueryStore.ts

```diff
@@ -0,0 +1,95 @@
+import {
+    AnyQuery,
+    isQueryParameter,
+    iterateRecursively,
+} from '@synthql/queries';
+import { SynthqlError } from './SynthqlError';
+
+export type QueryFunction = (...params: unknown[]) => AnyQuery;
+
+export class QueryStore {
+    private queries: Map<string, QueryFunction>;
+
+    constructor() {
+        this.queries = new Map();
+    }
+
+    /**
+     * Finds a query from the store and
+     * applies the parameters to the query.
+     *
+     * Throws an error if the query cannot be found,
+     * or if any of the parameters cannot be applied.
+     */
+    get({
+        queryId,
+        params,
+    }: {
+        queryId: string;
+        params: Record<string, unknown>;
+    }): AnyQuery {
+        const queryFn = this.queries.get(queryId);
+
+        if (!queryFn) {
+            throw SynthqlError.createQueryNotRegisteredError({ queryId });
+        }
+
+        const query = queryFn();
```

Comment:
Something is inconsistent here: the QueryFunction seems to take a list of params, but here you are passing none. Which one is it?

Reply:
I'm thinking: wWhy even have Query functions? why not store just regular queries?